frontend/src/plugin/Plugins.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import helpers from '../helpers';
+import { setPluginsLoadState } from '../redux/actions/actions';
+import { initializePlugins } from './index';
+
+/**
+ * For discovering and executing plugins.
+ */
+export default function Plugins() {
+  const dispatch = useDispatch();
+
+  // only run on first load
+  useEffect(() => {
+    const fetchData = async () => {
+      const pluginsScriptPaths = (await fetch(`${helpers.getAppUrl()}plugins/list`).then(resp =>
+        resp.json()
+      )) as string[];
+
+      const sources = await Promise.all(
+        pluginsScriptPaths.map(path =>
+          fetch(`${helpers.getAppUrl()}${path}`).then(resp => resp.text())
+        )
+      );
```

Comment:
Nice! I guess we get some speed improvements by having these (sort of) parallel.